src/app/core/products/product.service.ts

```diff
@@ -0,0 +1,124 @@
+            origin: "Colombia",
+            price: "$24.00",
+            process: "Honey",
+            varietal: "Geisha",
+            description: "Honey, Geisha",
+            tastingNotes: "White Flower, Pomegranate, Honeycomb, Raspberry",
+            collection: "featured",
+            type: "coffee"
+        },
+        {
+            id: 2,
+            name: "La Dalia",
+            origin: "Salvador",
+            price: "$15.00",
+            process: "48hrs Anaerobic",
+            varietal: "Pacamara",
+            description: "48hrs Anaerobic, Pacamara",
+            tastingNotes: "Green Apple, Red Currant, Strawberry",
+            collection: "upcoming",
+            type: "coffee"
+        },
+        {
+            id: 3,
+            name: "El Diviso",
+            origin: "Colombia",
+            price: "$16.00",
+            process: "Washed",
+            varietal: "Pink Bourbon",
+            description: "Washed, Pink Bourbon",
+            tastingNotes: "Grapefruit, Orange, Peach, Sugar",
+            collection: "upcoming",
+            type: "coffee"
+        },
+        {
+            id: 4,
+            name: "Rusatira",
+            origin: "Rwanda",
+            price: "$15.90",
+            process: "Wet Natural Anaerobic",
+            varietal: "Red Bourbon",
+            description: "Wet Natural Anaerobic, Red Bourbon",
+            tastingNotes: "Red Grapes, Peach, Nougat, Green Tea",
+            collection: undefined,
+            type: "coffee"
+        },
+        {
+            id: 5,
+            name: "Aurora",
+            origin: "Nicaragua",
+            price: "$11.00",
+            process: "Natural",
+            varietal: "Paraneima",
+            description: "Natural, Paraneima",
+            tastingNotes: "Hazelnut, Honey, Lime, Rose Hips, Stone Fruit",
+            collection: undefined,
+            type: "coffee"
+        },
+        {
+            id: 6,
+            name: "Nuna II",
+            origin: "Colombia",
+            price: "$14.00",
+            process: "Natural",
+            varietal: "Castillo",
+            description: "Natural, Castillo",
+            tastingNotes: "Black Cherry, Guava, Sugar",
+            collection: undefined,
+            type: "coffee"
+        },
+        {
+            id: 7,
+            name: "Orange Cream",
+            origin: "Colombia",
+            price: "$11.75",
+            process: "Washed",
+            varietal: "Caturra",
+            description: "Washed, Caturra",
+            tastingNotes: "Orange, Marshmallow, Pear, Cane Sugar",
+            collection: undefined,
+            type: "coffee"
+        },
+    ];
+
+
+    constructor() { }
+
+    ngOnInit() {
+    }
+
+    getProducts(): Observable<Coffee[]> {
+        return of(this.products);
+    }
+
+    getFeaturedProducts(): Observable<Coffee[]> {
+        return of(this.products
+            .filter(p => p.collection === "featured")
+            .slice(0, 2));
+    }
```

Comment:
this looks good in our case. However, we eventually want the data to be piped from the api directly in the filter. To make things easier for us, we should use the rxjs operator filter instead of the vanillajs array filter. That way, we can filter "on the fly" as values are received from an API. You can read up on it: https://www.learnrxjs.io/learn-rxjs/operators/filtering/filter

Here, we should call getProducts in getFeaturedProducts and then filter the observable (using rxjs) and return the resulting observable